src/shared/popups/gokb-edit-platform-popup/gokb-edit-platform-popup.vue

```diff
@@ -0,0 +1,230 @@
+                />
+              </v-col>
+            </v-row>
+          </div>
+        </template>
+      </v-col>
+    </v-row>
+    <v-row
+      dense
+      class="mt-4"
+    >
+      <v-col>
+        <gokb-title-field
+          v-model="platform.title"
+          :label="$tc('component.platform.label')"
+          required
+          dense
+          return-object
+        />
+      </v-col>
+    </v-row>
+    <v-row dense>
+      <v-col>
+        <gokb-url-field
+          ref="platformUrl"
+          v-model="platform.url"
+          :label="$tc('component.tipp.url.label')"
+          required
+          dense
+        />
+      </v-col>
+    </v-row>
+
+    <template #buttons>
+      <v-spacer />
+      <gokb-button
+        @click="close"
+      >
+        {{ $t('btn.cancel') }}
+      </gokb-button>
+      <gokb-button
+        v-if="!isReadonly"
+        :disabled="!isValid"
+        default
+      >
+        {{ isEdit ? $t('btn.update') : $t('btn.create') }}
+      </gokb-button>
+    </template>
+  </gokb-dialog>
+</template>
+
+<script>
+  import BaseComponent from '@/shared/components/base-component'
+  import accountModel from '@/shared/models/account-model'
+  import platformServices from '@/shared/services/platform-services'
+  import 'vue-json-pretty/lib/styles.css'
+
+  export default {
+    name: 'GokbEditPlatformPopup',
+    extends: BaseComponent,
+    props: {
+      selected: {
+        type: Object,
+        required: false,
+        default: undefined
+      },
+      readonly: {
+        type: Boolean,
+        required: false,
+        default: false
+      },
+      component: {
+        type: Object,
+        required: false,
+        default: undefined
+      }
+    },
+    data () {
+      return {
+        selectedItem: undefined,
+        successMsg: undefined,
+        errorMsg: undefined,
+        updateUrl: undefined,
+        platform: {
+          label: undefined,
+          url: undefined
+        },
+        items: []
+      }
+    },
+    computed: {
+      localValue: {
+        get () {
+          return this.value || true
+        },
+        set (localValue) {
+          this.$emit('input', localValue)
+        }
+      },
+      isReadonly () {
+        return !accountModel.loggedIn() || !accountModel.hasRole('ROLE_EDITOR') || (this.isEdit && !this.updateUrl)
+      },
+      isEdit () {
+        return !!this.id
+      },
+      isValid () {
+        return !!this.platformItem.component
+      },
+      localSuccessMessage () {
+        return this.successMsg ? this.$i18n.t(this.successMsg, [this.$i18n.tc('component.platform.label')]) : undefined
+      },
+      localErrorMessage () {
+        return this.errorMsg ? this.$i18n.t(this.errorMsg, [this.$i18n.tc('component.platform.label')]) : undefined
+      }
+    },
+    async created () {
+      this.selectedItem = this.selected
+
+      if (this.selected) {
+        this.id = this.selected.id
+        this.fetch(this.id)
```

Comment:
Fetching the single popup item again on creation is only necessary if there is some information not present in `this.selected` (aka the list entry). Otherwise, the properties from `this.selected` just have to be mapped onto our data object `platform`.

Reply:
... like it's done in the following lines: `if (this.component) {
        this.platform.component = this.component
      }` right, @hornmo ? I'm removing the lines you've commented on.